app/models/service_template_transformation_plan_task.rb

```diff
@@ -46,21 +49,74 @@ def task_active
     vm_resource.update_attributes(:status => ServiceResource::STATUS_ACTIVE)
   end
 
-  def conversion_host
-    Host.find_by(:id => options[:transformation_host_id])
+  def source_ems
+    options[:source_ems_id] ||= source.ext_management_system.id
+    source.ext_management_system
+  end
```

Comment:
Hm these look a little strange, combination of a setter and a getter in one method?

Reply:
Where are these used?  Could the caller set `options[:source_ems_id] = source_ems`?

Reply:
Could you drop the options setter part?  That's the only bit that I have an issue with and we can do that in a follow-up when we figure out where they are being used.